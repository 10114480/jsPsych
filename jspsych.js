// jspsych.js - a jQuery plugin for running psychology experiments
(function( $ ) {
	$.fn.jsPsych = function(options) {
		// build main options list before element iteration
		var opts = $.extend({}, $.fn.jsPsych.defaults, options);
		// exp structure
		var exp_blocks = [];
		var curr_block = 0;
		// execute experiment
		return this.each(function() {
			$this = $(this);
			
			// take the experiment structure and dynamically create a set of blocks
			exp_blocks = new Array(opts["experiment_structure"].length);
			
			for(var i = exp_blocks.length-1; i>=0; i--)
			{
				var trials = "undefined";
				
				for(var j = 0; j < opts["plugins"].length; j++)
				{
					if(opts["experiment_structure"][i]["type"] == opts["plugins"][j]["type"])
					{
						trials = opts["plugins"][j]["createFunc"].call(null, opts["experiment_structure"][i]);
					}
				}
				
				exp_blocks[i] = createBlock(trials);
				
			}

			// run the first block
			exp_blocks[0].next();
		});
		//
		// private functions //
		//
		function nextBlock()
		{
			curr_block += 1;
			if(curr_block == exp_blocks.length)
			{
				finishExperiment();
			} else {
				exp_blocks[curr_block].next();
			}
		}
		
		function createBlock(trial_list)
		{
			var block = {
				trial_idx: -1,
			
				trials: trial_list,
				
				data: [],
				
				next: function() {
					this.trial_idx = this.trial_idx+1;
				
					curr_trial = trial_list[this.trial_idx];
					
					if ( typeof curr_trial == "undefined"){
						return this.done();
					}
					
					do_trial(this, curr_trial);
				},
				
				done: nextBlock
			}
			
			return block;
		}
		
		function finishExperiment()
		{
			var all_data = [];
			for(var i=0;i<exp_blocks.length;i++)
			{
				all_data[i] = exp_blocks[i].data;
			}
			
			opts["finish"].apply((new Object()), [all_data]); 
		}
		
		function do_trial(block, trial)
		{
			for(var j = 0; j < opts["plugins"].length; j++)
			{
				if(trial.type == opts["plugins"][j]["type"])
				{
					opts["plugins"][j]["trialFunc"].call(this, $this, block, trial, 1);
				}
			}
		}
	};
	//
	// default parameters // 
	//
	$.fn.jsPsych.defaults = {
		// overall experiment parameters
		experiment_structure: [],
		plugins: [
				{"type": "sim",  "createFunc": similarity_create, "trialFunc": similarity_trial},
				{"type": "sd",   "createFunc": sd_create,         "trialFunc": sd_trial},
				{"type": "text", "createFunc": text_create,       "trialFunc": text_trial},
				{"type": "xab",  "createFunc": xab_create,        "trialFunc": xab_trial}]
	};
	//
	// useful helper functions for multiple plugins //
	//
	$.fn.jsPsych.showImage = function($this, img_path, img_class)
	{
		var the_img = document.createElement('img');
		the_img.setAttribute('src', img_path);
		the_img.setAttribute('class',img_class);
		$this.append(the_img);
	}

	$.fn.jsPsych.showImages = function($this, img_list, img_class)
	{
		for(img in img_list)
		{
			$.fn.jsPsych.showImage($this, img_list[img], img_class);
		}
	}
	
}) (jQuery);



