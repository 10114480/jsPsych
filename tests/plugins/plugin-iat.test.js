const root = '../../';

jest.useFakeTimers();

describe('iat plugin', function(){

  beforeEach(function(){
    require(root + 'jspsych.js');
    require(root + 'plugins/jspsych-iat.js');
  });

  test('loads correctly', function(){
    expect(typeof window.jsPsych.plugins['iat']).not.toBe('undefined');
  });

  test('displays image by default', function(){
    var trial = {
      type: 'iat',
      stimulus: '../media/blue.png'
    }

    jsPsych.init({
      timeline: [trial]
    });


    expect(jsPsych.getDisplayElement.innerHTML).toMatchSnapshot();
  });

  test('displays html when is_html is true', function(){
    var trial = {
      type: 'iat',
      stimulus: '<p>hello</p>',
      is_html: true
    }

    jsPsych.init({
      timeline: [trial]
    });

    expect(jsPsych.getDisplayElement().innerHTML).toMatchSnapshot();
  });

  test('display should only clear when left key is pressed', function(){
    var trial = {
      type: 'iat',
      stimulus: '<p>hello</p>',
      is_html: true,
      left_category_key: 'f',
      left_category_label: ['FRIENDLY'],
      stim_key_association: 'left',
      key_to_move_forward: ['f']
    }

    jsPsych.init({
      timeline: [trial]
    });

    document.dispatchEvent(new KeyboardEvent('keydown', {keyCode: 32}));
    document.dispatchEvent(new KeyboardEvent('keyup', {keyCode: 32}));

    expect(jsPsych.getDisplayElement().innerHTML).toMatchSnapshot();

    document.dispatchEvent(new KeyboardEvent('keydown', {keyCode: 70}));
    document.dispatchEvent(new KeyboardEvent('keyup', {keyCode: 70}));

    expect(jsPsych.getDisplayElement().innerHTML).toMatchSnapshot();
  });

  test('display should only clear when right key is pressed', function(){
    var trial = {
      type: 'iat',
      stimulus: '<p>hello</p>',
      is_html: true,
      right_category_key: 'j',
      right_category_label: ['UNFRIENDLY'],
      stim_key_association: 'right',
      key_to_move_forward: ['j']
    }

    jsPsych.init({
      timeline: [trial]
    });

    document.dispatchEvent(new KeyboardEvent('keydown', {keyCode: 32}));
    document.dispatchEvent(new KeyboardEvent('keyup', {keyCode: 32}));

    expect(jsPsych.getDisplayElement().innerHTML).toMatchSnapshot();

    document.dispatchEvent(new KeyboardEvent('keydown', {keyCode: 74}));
    document.dispatchEvent(new KeyboardEvent('keyup', {keyCode: 74}));

    expect(jsPsych.getDisplayElement().innerHTML).toMatchSnapshot();

  });

  test('display should clear when any key is pressed', function(){
    var trial = {
      type: 'iat',
      stimulus: '<p>hello</p>',
      is_html: true,
      right_category_key: 'j',
      right_category_label: ['UNFRIENDLY'],
      stim_key_association: 'right',
      key_to_move_forward: [jsPsych.ALL_KEYS]
    }

    jsPsych.init({
      timeline: [trial]
    });

    document.dispatchEvent(new KeyboardEvent('keydown', {keyCode: 32}));
    document.dispatchEvent(new KeyboardEvent('keyup', {keyCode: 32}));

    expect(jsPsych.getDisplayElement().innerHTML).toMatchSnapshot();

  });

  test('display should only when "other key" is pressed', function(){
    var trial = {
      type: 'iat',
      stimulus: '<p>hello</p>',
      is_html: true,
      left_category_key: 'f',
      right_category_key: 'j',
      left_category_label: ['FRIENDLY'],
      right_category_label: ['UNFRIENDLY'],
      stim_key_association: 'left',
      key_to_move_forward: ['other key']
    }

    jsPsych.init({
      timeline: [trial]
    });

    document.dispatchEvent(new KeyboardEvent('keydown', {keyCode: 74}));
    document.dispatchEvent(new KeyboardEvent('keyup', {keyCode: 74}));

    expect(jsPsych.getDisplayElement().innerHTML).toMatchSnapshot();

    document.dispatchEvent(new KeyboardEvent('keydown', {keyCode: 70}));
    document.dispatchEvent(new KeyboardEvent('keyup', {keyCode: 70}));

    expect(jsPsych.getDisplayElement().innerHTML).toMatchSnapshot();

  });


  test('display should clear if key press is associated with category label', function(){
    var trial = {
      type: 'iat',
      stimulus: '<p>hello</p>',
      is_html: true,
      left_category_key: 'f',
      right_category_key: 'j',
      left_category_label: ['FRIENDLY'],
      right_category_label: ['UNFRIENDLY'],
      stim_key_association: 'left'
    }

    jsPsych.init({
      timeline: [trial]
    });

    document.dispatchEvent(new KeyboardEvent('keydown', {keyCode:70}));
    document.dispatchEvent(new KeyboardEvent('keyup', {keyCode:70}));

    expect(jsPsych.getDisplayElement().innerHTML).toMatchSnapshot();
  });

  test('should display wrong image when wrong key is pressed', function(){
    var trial = {
      type: 'iat',
      stimulus: '<p>hello</p>',
      image_when_wrong: '../media/redX.png',
      wrong_image_name: 'red X',
      is_html: true,
      left_category_key: 'f',
      right_category_key: 'j',
      left_category_label: ['FRIENDLY'],
      right_category_label: ['UNFRIENDLY'],
      stim_key_association: 'left',
      key_to_move_forward: ['f']
    }

    jsPsych.init({
      timeline: [trial]
    });

    document.dispatchEvent(new KeyboardEvent('keydown', {keyCode:74}));
    document.dispatchEvent(new KeyboardEvent('keyup', {keyCode:74}));

    expect(jsPsych.getDisplayElement().innerHTML).toMatchSnapshot();
  });

  test('when wrong image is displayed, display should clear after pressing key to move forward', function(){
    var trial = {
      type: 'iat',
      stimulus: '<p>hello</p>',
      image_when_wrong: '../media/redX.png',
      wrong_image_name: 'red X',
      is_html: true,
      left_category_key: 'f',
      right_category_key: 'j',
      left_category_label: ['FRIENDLY'],
      right_category_label: ['UNFRIENDLY'],
      stim_key_association: 'left',
      key_to_move_forward: ['f']
    }

    jsPsych.init({
      timeline: [trial]
    });

    document.dispatchEvent(new KeyboardEvent('keydown', {keyCode:74}));
    document.dispatchEvent(new KeyboardEvent('keyup', {keyCode:74}));

    expect(jsPsych.getDisplayElement().innerHTML).toMatchSnapshot();

    document.dispatchEvent(new KeyboardEvent('keydown', {keyCode:70}));
    document.dispatchEvent(new KeyboardEvent('keyup', {keyCode:70}));

    expect(jsPsych.getDisplayElement().innerHTML).toMatchSnapshot();
  });


  test('prompt should append html below stimulus', function(){
    var trial = {
      type: 'iat',
      stimulus: '<p>hello</p>',
      is_html: true,
      prompt: '<div id="foo">this is the prompt</div>'
    }

    jsPsych.init({
      timeline: [trial]
    });

    expect(jsPsych.getDisplayElement().innerHTML).toMatchSnapshot();

    document.dispatchEvent(new KeyboardEvent('keydown', {keyCode: 70}));
    document.dispatchEvent(new KeyboardEvent('keyup', {keyCode: 70}));

  });

  test('timing_response should end trial after time has elapsed; only if display_feedback is false', function(){

    var trial = {
      type: 'iat',
      stimulus: '<p>hello</p>',
      is_html: true,
      display_feedback: false,
      timing_response: 500
    }

    jsPsych.init({
      timeline: [trial]
    });

    expect(jsPsych.getDisplayElement().innerHTML).toMatchSnapshot();

    jest.runAllTimers();

    expect(jsPsych.getDisplayElement().innerHTML).toMatchSnapshot();

  });

  test('trial should not end when response_ends_trial is false and stimulus should get responded class', function(){
    var trial = {
      type: 'iat',
      stimulus: '<p>hello</p>',
      is_html: true,
      response_ends_trial: false,
      display_feedback: false,
      timing_response: 500
    }

    jsPsych.init({
      timeline: [trial]
    });

    document.dispatchEvent(new KeyboardEvent('keydown', {keyCode: 70}));
    document.dispatchEvent(new KeyboardEvent('keyup', {keyCode: 70}));

    expect(jsPsych.getDisplayElement().innerHTML).toMatchSnapshot();

    jest.runAllTimers();
  });

  test('should accept functions as parameters(timing_response in use)', function(){

    var trial = {
      type: 'iat',
      stimulus: function(){ return '<p>hello</p>'; },
      is_html: function(){ return true; },
      display_feedback: function(){ return false; },
      image_when_wrong: function(){ return '../media/redX.png'; },
      wrong_image_name: function(){return 'red X'; },
      left_category_key: function(){ return 'E'; },
      right_category_key: function(){ return 'I'; },
      left_category_label: function(){return ['FRIENDLY']; },
      right_category_label: function(){return ['UNFRIENDLY']; },
      stim_key_association: function(){return 'left'; },
      prompt: function(){ return '<div>prompt</div>'; },
      timing_response: function(){ return 1000; },
      response_ends_trial: function(){ return false; }
    }

    jsPsych.init({
      timeline: [trial]
    });

    expect(jsPsych.getDisplayElement().innerHTML).toMatchSnapshot();

    document.dispatchEvent(new KeyboardEvent('keydown', {keyCode: 32}));
    document.dispatchEvent(new KeyboardEvent('keyup', {keyCode: 32}));

    expect(jsPsych.getDisplayElement().innerHTML).toMatchSnapshot();

    document.dispatchEvent(new KeyboardEvent('keydown', {keyCode: 70}));
    document.dispatchEvent(new KeyboardEvent('keyup', {keyCode: 70}));

    expect(jsPsych.getDisplayElement().innerHTML).toMatchSnapshot();

    jest.runTimersToTime(500);

    expect(jsPsych.getDisplayElement().innerHTML).toMatchSnapshot();

    jest.runTimersToTime(1000);

    expect(jsPsych.getDisplayElement().innerHTML).toMatchSnapshot();
  });

  test('should accept functions as parameters(timing_response is not in use)', function(){

    var trial = {
      type: 'iat',
      stimulus: function(){ return '<p>hello</p>'; },
      is_html: function(){ return true; },
      display_feedback: function(){ return true; },
      image_when_wrong: function(){ return '../media/redX.png'; },
      wrong_image_name: function(){return 'red X'; },
      left_category_key: function(){ return 'E'; },
      right_category_key: function(){ return 'I'; },
      left_category_label: function(){return ['FRIENDLY']; },
      right_category_label: function(){return ['UNFRIENDLY']; },
      stim_key_association: function(){return 'left'; },
      key_to_move_forward: function(){return ['other key']; },
      prompt: function(){ return '<div>prompt</div>'; },
      timing_response: function(){ return 1000; },
      response_ends_trial: function(){ return false; }
    }

    jsPsych.init({
      timeline: [trial]
    });

    expect(jsPsych.getDisplayElement().innerHTML).toMatchSnapshot();

    document.dispatchEvent(new KeyboardEvent('keydown', {keyCode: 32}));
    document.dispatchEvent(new KeyboardEvent('keyup', {keyCode: 32}));

    expect(jsPsych.getDisplayElement().innerHTML).toMatchSnapshot();

    document.dispatchEvent(new KeyboardEvent('keydown', {keyCode: 70}));
    document.dispatchEvent(new KeyboardEvent('keyup', {keyCode: 70}));

    expect(jsPsych.getDisplayElement().innerHTML).toMatchSnapshot();

    jest.runTimersToTime(500);

    expect(jsPsych.getDisplayElement().innerHTML).toMatchSnapshot();

    jest.runTimersToTime(1000);

    expect(jsPsych.getDisplayElement().innerHTML).toMatchSnapshot();
  });
});