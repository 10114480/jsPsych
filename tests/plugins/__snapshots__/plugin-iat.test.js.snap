// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`iat plugin display should clear if key press is associated with category label 1`] = `""`;

exports[`iat plugin display should clear when any key is pressed 1`] = `" <div style=\\"position: absolute; height: 20%; width: 100%; margin-left: auto; margin-right: auto; top: 42%; left: 0; right: 0\\"><p id=\\"jspsych-iat-stim\\"></p><p>hello</p><p></p></div><div id=\\"trial_left_align\\" style=\\"position: absolute; top: 18%; left: 20%\\"><p>Press E for:<br> <b>left</b></p></div><div id=\\"trial_right_align\\" style=\\"position: absolute; top: 18%; right: 20%\\"><p>Press j for:<br> <b>UNFRIENDLY</b></p></div><div id=\\"wrongImg\\" style=\\"position:relative; top: 300px; margin-left: auto; margin-right: auto; left: 0; right: 0\\"><p>Trial will continue automatically.</p></div>"`;

exports[`iat plugin display should only clear when left key is pressed 1`] = `" <div style=\\"position: absolute; height: 20%; width: 100%; margin-left: auto; margin-right: auto; top: 42%; left: 0; right: 0\\"><p id=\\"jspsych-iat-stim\\"></p><p>hello</p><p></p></div><div id=\\"trial_left_align\\" style=\\"position: absolute; top: 18%; left: 20%\\"><p>Press f for:<br> <b>FRIENDLY</b></p></div><div id=\\"trial_right_align\\" style=\\"position: absolute; top: 18%; right: 20%\\"><p>Press I for:<br> <b>right</b></p></div><div id=\\"wrongImg\\" style=\\"position:relative; top: 300px; margin-left: auto; margin-right: auto; left: 0; right: 0\\"><p>Trial will continue automatically.</p></div>"`;

exports[`iat plugin display should only clear when left key is pressed 2`] = `""`;

exports[`iat plugin display should only clear when right key is pressed 1`] = `" <div style=\\"position: absolute; height: 20%; width: 100%; margin-left: auto; margin-right: auto; top: 42%; left: 0; right: 0\\"><p id=\\"jspsych-iat-stim\\"></p><p>hello</p><p></p></div><div id=\\"trial_left_align\\" style=\\"position: absolute; top: 18%; left: 20%\\"><p>Press E for:<br> <b>left</b></p></div><div id=\\"trial_right_align\\" style=\\"position: absolute; top: 18%; right: 20%\\"><p>Press j for:<br> <b>UNFRIENDLY</b></p></div><div id=\\"wrongImg\\" style=\\"position:relative; top: 300px; margin-left: auto; margin-right: auto; left: 0; right: 0\\"><p>Trial will continue automatically.</p></div>"`;

exports[`iat plugin display should only clear when right key is pressed 2`] = `""`;

exports[`iat plugin display should only when "other key" is pressed 1`] = `" <div style=\\"position: absolute; height: 20%; width: 100%; margin-left: auto; margin-right: auto; top: 42%; left: 0; right: 0\\"><p id=\\"jspsych-iat-stim\\"></p><p>hello</p><p></p></div><div id=\\"trial_left_align\\" style=\\"position: absolute; top: 18%; left: 20%\\"><p>Press f for:<br> <b>FRIENDLY</b></p></div><div id=\\"trial_right_align\\" style=\\"position: absolute; top: 18%; right: 20%\\"><p>Press j for:<br> <b>UNFRIENDLY</b></p></div><div id=\\"wrongImg\\" style=\\"position:relative; top: 300px; margin-left: auto; margin-right: auto; left: 0; right: 0\\"><p>Trial will continue automatically.</p></div>"`;

exports[`iat plugin display should only when "other key" is pressed 2`] = `" <div style=\\"position: absolute; height: 20%; width: 100%; margin-left: auto; margin-right: auto; top: 42%; left: 0; right: 0\\"><p id=\\"jspsych-iat-stim\\" class=\\" responded\\"></p><p>hello</p><p></p></div><div id=\\"trial_left_align\\" style=\\"position: absolute; top: 18%; left: 20%\\"><p>Press f for:<br> <b>FRIENDLY</b></p></div><div id=\\"trial_right_align\\" style=\\"position: absolute; top: 18%; right: 20%\\"><p>Press j for:<br> <b>UNFRIENDLY</b></p></div><div id=\\"wrongImg\\" style=\\"position:relative; top: 300px; margin-left: auto; margin-right: auto; left: 0; right: 0\\"><p>Trial will continue automatically.</p></div>"`;

exports[`iat plugin displays html when is_html is true 1`] = `" <div style=\\"position: absolute; height: 20%; width: 100%; margin-left: auto; margin-right: auto; top: 42%; left: 0; right: 0\\"><p id=\\"jspsych-iat-stim\\"></p><p>hello</p><p></p></div><div id=\\"trial_left_align\\" style=\\"position: absolute; top: 18%; left: 20%\\"><p>Press E for:<br> <b>left</b></p></div><div id=\\"trial_right_align\\" style=\\"position: absolute; top: 18%; right: 20%\\"><p>Press I for:<br> <b>right</b></p></div><div id=\\"wrongImg\\" style=\\"position:relative; top: 300px; margin-left: auto; margin-right: auto; left: 0; right: 0\\"><p>Trial will continue automatically.</p></div>"`;

exports[`iat plugin displays image by default 1`] = `undefined`;

exports[`iat plugin prompt should append html below stimulus 1`] = `" <div style=\\"position: absolute; height: 20%; width: 100%; margin-left: auto; margin-right: auto; top: 42%; left: 0; right: 0\\"><p id=\\"jspsych-iat-stim\\"></p><p>hello</p><p></p></div><div id=\\"trial_left_align\\" style=\\"position: absolute; top: 18%; left: 20%\\"><p>Press E for:<br> <b>left</b></p></div><div id=\\"trial_right_align\\" style=\\"position: absolute; top: 18%; right: 20%\\"><p>Press I for:<br> <b>right</b></p></div><div id=\\"wrongImg\\" style=\\"position:relative; top: 300px; margin-left: auto; margin-right: auto; left: 0; right: 0\\"><p>Trial will continue automatically.</p></div><div id=\\"foo\\">this is the prompt</div>"`;

exports[`iat plugin should accept functions as parameters(timing_response in use) 1`] = `" <div style=\\"position: absolute; height: 20%; width: 100%; margin-left: auto; margin-right: auto; top: 42%; left: 0; right: 0\\"><p id=\\"jspsych-iat-stim\\"></p><p>hello</p><p></p></div><div id=\\"trial_left_align\\" style=\\"position: absolute; top: 18%; left: 20%\\"><p>Press E for:<br> <b>FRIENDLY</b></p></div><div id=\\"trial_right_align\\" style=\\"position: absolute; top: 18%; right: 20%\\"><p>Press I for:<br> <b>UNFRIENDLY</b></p></div><div id=\\"wrongImg\\" style=\\"position:relative; top: 300px; margin-left: auto; margin-right: auto; left: 0; right: 0\\"><p>Trial will continue automatically.</p></div><div>prompt</div>"`;

exports[`iat plugin should accept functions as parameters(timing_response in use) 2`] = `" <div style=\\"position: absolute; height: 20%; width: 100%; margin-left: auto; margin-right: auto; top: 42%; left: 0; right: 0\\"><p id=\\"jspsych-iat-stim\\"></p><p>hello</p><p></p></div><div id=\\"trial_left_align\\" style=\\"position: absolute; top: 18%; left: 20%\\"><p>Press E for:<br> <b>FRIENDLY</b></p></div><div id=\\"trial_right_align\\" style=\\"position: absolute; top: 18%; right: 20%\\"><p>Press I for:<br> <b>UNFRIENDLY</b></p></div><div id=\\"wrongImg\\" style=\\"position:relative; top: 300px; margin-left: auto; margin-right: auto; left: 0; right: 0\\"><p>Trial will continue automatically.</p></div><div>prompt</div>"`;

exports[`iat plugin should accept functions as parameters(timing_response in use) 3`] = `" <div style=\\"position: absolute; height: 20%; width: 100%; margin-left: auto; margin-right: auto; top: 42%; left: 0; right: 0\\"><p id=\\"jspsych-iat-stim\\"></p><p>hello</p><p></p></div><div id=\\"trial_left_align\\" style=\\"position: absolute; top: 18%; left: 20%\\"><p>Press E for:<br> <b>FRIENDLY</b></p></div><div id=\\"trial_right_align\\" style=\\"position: absolute; top: 18%; right: 20%\\"><p>Press I for:<br> <b>UNFRIENDLY</b></p></div><div id=\\"wrongImg\\" style=\\"position:relative; top: 300px; margin-left: auto; margin-right: auto; left: 0; right: 0\\"><p>Trial will continue automatically.</p></div><div>prompt</div>"`;

exports[`iat plugin should accept functions as parameters(timing_response in use) 4`] = `" <div style=\\"position: absolute; height: 20%; width: 100%; margin-left: auto; margin-right: auto; top: 42%; left: 0; right: 0\\"><p id=\\"jspsych-iat-stim\\"></p><p>hello</p><p></p></div><div id=\\"trial_left_align\\" style=\\"position: absolute; top: 18%; left: 20%\\"><p>Press E for:<br> <b>FRIENDLY</b></p></div><div id=\\"trial_right_align\\" style=\\"position: absolute; top: 18%; right: 20%\\"><p>Press I for:<br> <b>UNFRIENDLY</b></p></div><div id=\\"wrongImg\\" style=\\"position:relative; top: 300px; margin-left: auto; margin-right: auto; left: 0; right: 0\\"><p>Trial will continue automatically.</p></div><div>prompt</div>"`;

exports[`iat plugin should accept functions as parameters(timing_response in use) 5`] = `""`;

exports[`iat plugin should accept functions as parameters(timing_response is not in use) 1`] = `" <div style=\\"position: absolute; height: 20%; width: 100%; margin-left: auto; margin-right: auto; top: 42%; left: 0; right: 0\\"><p id=\\"jspsych-iat-stim\\"></p><p>hello</p><p></p></div><div id=\\"trial_left_align\\" style=\\"position: absolute; top: 18%; left: 20%\\"><p>Press E for:<br> <b>FRIENDLY</b></p></div><div id=\\"trial_right_align\\" style=\\"position: absolute; top: 18%; right: 20%\\"><p>Press I for:<br> <b>UNFRIENDLY</b></p></div><div id=\\"wrongImg\\" style=\\"position:relative; top: 300px; margin-left: auto; margin-right: auto; left: 0; right: 0\\"><div id=\\"wrongImgContainer\\" style=\\"position: absolute; top: -75px; margin-left: auto; margin-right: auto; left: 0; right: 0\\"><img src=\\"../media/redX.png\\" style=\\"visibility: hidden;\\" id=\\"wrongImgID\\"></div><p>If you press the wrong key, a red X will appear. Press the other key to continue.</p></div><div>prompt</div>"`;

exports[`iat plugin should accept functions as parameters(timing_response is not in use) 2`] = `" <div style=\\"position: absolute; height: 20%; width: 100%; margin-left: auto; margin-right: auto; top: 42%; left: 0; right: 0\\"><p id=\\"jspsych-iat-stim\\"></p><p>hello</p><p></p></div><div id=\\"trial_left_align\\" style=\\"position: absolute; top: 18%; left: 20%\\"><p>Press E for:<br> <b>FRIENDLY</b></p></div><div id=\\"trial_right_align\\" style=\\"position: absolute; top: 18%; right: 20%\\"><p>Press I for:<br> <b>UNFRIENDLY</b></p></div><div id=\\"wrongImg\\" style=\\"position:relative; top: 300px; margin-left: auto; margin-right: auto; left: 0; right: 0\\"><div id=\\"wrongImgContainer\\" style=\\"position: absolute; top: -75px; margin-left: auto; margin-right: auto; left: 0; right: 0\\"><img src=\\"../media/redX.png\\" style=\\"visibility: hidden;\\" id=\\"wrongImgID\\"></div><p>If you press the wrong key, a red X will appear. Press the other key to continue.</p></div><div>prompt</div>"`;

exports[`iat plugin should accept functions as parameters(timing_response is not in use) 3`] = `" <div style=\\"position: absolute; height: 20%; width: 100%; margin-left: auto; margin-right: auto; top: 42%; left: 0; right: 0\\"><p id=\\"jspsych-iat-stim\\"></p><p>hello</p><p></p></div><div id=\\"trial_left_align\\" style=\\"position: absolute; top: 18%; left: 20%\\"><p>Press E for:<br> <b>FRIENDLY</b></p></div><div id=\\"trial_right_align\\" style=\\"position: absolute; top: 18%; right: 20%\\"><p>Press I for:<br> <b>UNFRIENDLY</b></p></div><div id=\\"wrongImg\\" style=\\"position:relative; top: 300px; margin-left: auto; margin-right: auto; left: 0; right: 0\\"><div id=\\"wrongImgContainer\\" style=\\"position: absolute; top: -75px; margin-left: auto; margin-right: auto; left: 0; right: 0\\"><img src=\\"../media/redX.png\\" style=\\"visibility: hidden;\\" id=\\"wrongImgID\\"></div><p>If you press the wrong key, a red X will appear. Press the other key to continue.</p></div><div>prompt</div>"`;

exports[`iat plugin should accept functions as parameters(timing_response is not in use) 4`] = `" <div style=\\"position: absolute; height: 20%; width: 100%; margin-left: auto; margin-right: auto; top: 42%; left: 0; right: 0\\"><p id=\\"jspsych-iat-stim\\"></p><p>hello</p><p></p></div><div id=\\"trial_left_align\\" style=\\"position: absolute; top: 18%; left: 20%\\"><p>Press E for:<br> <b>FRIENDLY</b></p></div><div id=\\"trial_right_align\\" style=\\"position: absolute; top: 18%; right: 20%\\"><p>Press I for:<br> <b>UNFRIENDLY</b></p></div><div id=\\"wrongImg\\" style=\\"position:relative; top: 300px; margin-left: auto; margin-right: auto; left: 0; right: 0\\"><div id=\\"wrongImgContainer\\" style=\\"position: absolute; top: -75px; margin-left: auto; margin-right: auto; left: 0; right: 0\\"><img src=\\"../media/redX.png\\" style=\\"visibility: hidden;\\" id=\\"wrongImgID\\"></div><p>If you press the wrong key, a red X will appear. Press the other key to continue.</p></div><div>prompt</div>"`;

exports[`iat plugin should accept functions as parameters(timing_response is not in use) 5`] = `" <div style=\\"position: absolute; height: 20%; width: 100%; margin-left: auto; margin-right: auto; top: 42%; left: 0; right: 0\\"><p id=\\"jspsych-iat-stim\\"></p><p>hello</p><p></p></div><div id=\\"trial_left_align\\" style=\\"position: absolute; top: 18%; left: 20%\\"><p>Press E for:<br> <b>FRIENDLY</b></p></div><div id=\\"trial_right_align\\" style=\\"position: absolute; top: 18%; right: 20%\\"><p>Press I for:<br> <b>UNFRIENDLY</b></p></div><div id=\\"wrongImg\\" style=\\"position:relative; top: 300px; margin-left: auto; margin-right: auto; left: 0; right: 0\\"><div id=\\"wrongImgContainer\\" style=\\"position: absolute; top: -75px; margin-left: auto; margin-right: auto; left: 0; right: 0\\"><img src=\\"../media/redX.png\\" style=\\"visibility: hidden;\\" id=\\"wrongImgID\\"></div><p>If you press the wrong key, a red X will appear. Press the other key to continue.</p></div><div>prompt</div>"`;

exports[`iat plugin should display wrong image when wrong key is pressed 1`] = `" <div style=\\"position: absolute; height: 20%; width: 100%; margin-left: auto; margin-right: auto; top: 42%; left: 0; right: 0\\"><p id=\\"jspsych-iat-stim\\" class=\\" responded\\"></p><p>hello</p><p></p></div><div id=\\"trial_left_align\\" style=\\"position: absolute; top: 18%; left: 20%\\"><p>Press f for:<br> <b>FRIENDLY</b></p></div><div id=\\"trial_right_align\\" style=\\"position: absolute; top: 18%; right: 20%\\"><p>Press j for:<br> <b>UNFRIENDLY</b></p></div><div id=\\"wrongImg\\" style=\\"position:relative; top: 300px; margin-left: auto; margin-right: auto; left: 0; right: 0\\"><p>Trial will continue automatically.</p></div>"`;

exports[`iat plugin timing_response should end trial after time has elapsed; only if display_feedback is false 1`] = `" <div style=\\"position: absolute; height: 20%; width: 100%; margin-left: auto; margin-right: auto; top: 42%; left: 0; right: 0\\"><p id=\\"jspsych-iat-stim\\"></p><p>hello</p><p></p></div><div id=\\"trial_left_align\\" style=\\"position: absolute; top: 18%; left: 20%\\"><p>Press E for:<br> <b>left</b></p></div><div id=\\"trial_right_align\\" style=\\"position: absolute; top: 18%; right: 20%\\"><p>Press I for:<br> <b>right</b></p></div><div id=\\"wrongImg\\" style=\\"position:relative; top: 300px; margin-left: auto; margin-right: auto; left: 0; right: 0\\"><p>Trial will continue automatically.</p></div>"`;

exports[`iat plugin timing_response should end trial after time has elapsed; only if display_feedback is false 2`] = `""`;

exports[`iat plugin trial should not end when response_ends_trial is false and stimulus should get responded class 1`] = `" <div style=\\"position: absolute; height: 20%; width: 100%; margin-left: auto; margin-right: auto; top: 42%; left: 0; right: 0\\"><p id=\\"jspsych-iat-stim\\"></p><p>hello</p><p></p></div><div id=\\"trial_left_align\\" style=\\"position: absolute; top: 18%; left: 20%\\"><p>Press E for:<br> <b>left</b></p></div><div id=\\"trial_right_align\\" style=\\"position: absolute; top: 18%; right: 20%\\"><p>Press I for:<br> <b>right</b></p></div><div id=\\"wrongImg\\" style=\\"position:relative; top: 300px; margin-left: auto; margin-right: auto; left: 0; right: 0\\"><p>Trial will continue automatically.</p></div>"`;

exports[`iat plugin when wrong image is displayed, display should clear after pressing key to move forward 1`] = `" <div style=\\"position: absolute; height: 20%; width: 100%; margin-left: auto; margin-right: auto; top: 42%; left: 0; right: 0\\"><p id=\\"jspsych-iat-stim\\"></p><p>hello</p><p></p></div><div id=\\"trial_left_align\\" style=\\"position: absolute; top: 18%; left: 20%\\"><p>Press f for:<br> <b>FRIENDLY</b></p></div><div id=\\"trial_right_align\\" style=\\"position: absolute; top: 18%; right: 20%\\"><p>Press j for:<br> <b>UNFRIENDLY</b></p></div><div id=\\"wrongImg\\" style=\\"position:relative; top: 300px; margin-left: auto; margin-right: auto; left: 0; right: 0\\"><p>Trial will continue automatically.</p></div>"`;

exports[`iat plugin when wrong image is displayed, display should clear after pressing key to move forward 2`] = `" <div style=\\"position: absolute; height: 20%; width: 100%; margin-left: auto; margin-right: auto; top: 42%; left: 0; right: 0\\"><p id=\\"jspsych-iat-stim\\" class=\\" responded\\"></p><p>hello</p><p></p></div><div id=\\"trial_left_align\\" style=\\"position: absolute; top: 18%; left: 20%\\"><p>Press f for:<br> <b>FRIENDLY</b></p></div><div id=\\"trial_right_align\\" style=\\"position: absolute; top: 18%; right: 20%\\"><p>Press j for:<br> <b>UNFRIENDLY</b></p></div><div id=\\"wrongImg\\" style=\\"position:relative; top: 300px; margin-left: auto; margin-right: auto; left: 0; right: 0\\"><p>Trial will continue automatically.</p></div>"`;
